gammapy/cube/psf_map.py

```diff
@@ -331,3 +333,46 @@ def stack(self, other):
     def copy(self):
         """Copy PSFMap"""
         return deepcopy(self)
+
+    def sample_coord(psf_map, map_coord, random_state=0):
+        """ Apply PSF corrections on the coordinates of a set of simulated events.
+        psf_map : `~gammapy.cube.psf_map.PSFMap`.
+            Psf map at the position of each event.
+        map_coord : `~gammapy.maps.MapCoord` object.
+            Sequence of coordinates and energies of sampled events.
+        random_state : {int, 'random-seed', 'global-rng', `~numpy.random.RandomState`}
+            Defines random number generator initialisation.
+            Passed to `~gammapy.utils.random.get_random_state`.
+
+        Returns
+        -------
+        map_coord : `~gammapy.maps.MapCoord` object.
+            Sequence of PSF-corrected coordinates of the input map_coord map.
+        """
+
+        random_state = get_random_state(random_state)
+
+        map_coord["energy"].sort()
+        coord = {
+            "lon": map_coord.lon.reshape(-1, 1),
+            "lat": map_coord.lat.reshape(-1, 1),
+            "energy": map_coord["energy"].reshape(-1, 1),
+            "theta": (psf_map.psf_map.geom.axes[0].center),
+        }
+
+        pdf = psf_map.psf_map.interp_by_coord(coord)
+
+        sample_pdf = InverseCDFSampler(pdf, axis=1)
+        pix_coord = sample_pdf.sample_axis()
+        offset = psf_map.psf_map.geom.axes[0].pix_to_coord(pix_coord)
+
+        phi_angle = random_state.uniform(360, size=len(map_coord.lon)) * u.deg
+        event_positions = SkyCoord(
+            map_coord.lon, map_coord.lat, frame="galactic", unit="deg"
+        )
+        event_positions = event_positions.directional_offset_by(phi_angle, offset)
+
+        map_coord.lon[:] = event_positions.l
```

Comment:
I think we should not overwrite the input coordinates, but just create a new `MapCoord` object, including the reconstructed positions.